### ISSUE TITLE
yo jhipster:upgrade failed from 4.2 to 4.3

### ISSUE THREAD

**Issue by glxiong** [jhipster/generator-jhipster#5649]
Here is the output:

Welcome to the JHipster Upgrade Sub-Generator
This will upgrade your current application codebase to the latest JHipster version
Looking for latest generator-jhipster version...
New generator-jhipster version found: 4.3.0
Git repository detected
Created branch jhipster_upgrade
Cleaned up directory
Installing JHipster 4.2.0 locally
Installed generator-jhipster@4.2.0
Regenerating app with jhipster 4.2.0...
Successfully regenerated app with jhipster 4.2.0
Committed with message "Generated with JHipster 4.2.0"
Checked out branch "master"
Current code recorded as generated with version 4.2.0
Updating generator-jhipster to 4.3.0 . This might take some time...
Updated generator-jhipster to version 4.3.0
Checked out branch "jhipster_upgrade"
Cleaned up directory
Regenerating app with jhipster 4.3.0...
Successfully regenerated app with jhipster 4.3.0
Committed with message "Generated with JHipster 4.3.0"
Checked out branch "master"
Merging changes back to master...
Merge done!
Installing dependencies, please wait...
Error jhipster:upgrade 

ERROR! yarnfailed.

The yo jhipster:info gives the following output:

Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
smart-price@0.0.0 /Users/testuser/Demo/SmartPrice
└── (empty)

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

```yaml
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.company.smartprice"
    },
    "jhipsterVersion": "4.2.0",
    "baseName": "SmartPrice",
    "packageName": "com.company.smartprice",
    "packageFolder": "com/company/smartprice",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "286e201fdd317cb4c624967b490c25997edc8f0b",
    "clientFramework": "angular2",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

ls: no such file or directory: .jhipster/*.json

##### **Browsers and Operating System**

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

git version 2.11.0 (Apple Git-81)

node: v7.9.0

npm: 4.2.0

bower: 1.8.0

gulp:
[13:30:37] CLI version 1.2.2

yeoman: 1.8.5

yarn: 0.23.2

**Comment by jdubois** [comment]
You had an error while running Yarn, didn't you have any error message? Anything can happen here: it depends on your configuration, but also it can fail because of an Internet error, etc...

**Comment by cloud4288** [comment]
@glxiong I also encountered this error when I upgrade to 4.3.0. Just ignore in and fix any merge conflicts and run the yarn command.

**Comment by jdubois** [comment]
I confirm the issue

**Comment by jdubois** [comment]
This seems to come from `yarn.lock`, that is forcing to keep older version of dependencies (like Typescript, in this case). We should do a `yarn upgrade`, which is totally logical, in fact.

**Comment by hugotown** [comment]
Hi!
I have the same error when i trying to upgrade my jhipster project
Meanwhile how can we upgrade the project ?
regards!

**Comment by jdubois** [comment]
There are instructions to do it manually, you should follow them.
I'm sorry I didn't have the time to inspect this, and we need as much
information and help as possible here.
For example, you could give your version number, logs, etc... Just saying
that it doesn't work isn't making things go forward.


Le 11 mai 2017 4:00 AM, "Hugo Ruíz" <notifications@github.com> a écrit :

Hi!
I have the same error when i trying to upgrade my jhipster project
Meanwhile how can we upgrade the project ?
regards!

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/jhipster/generator-jhipster/issues/5649#issuecomment-300661421>,
or mute the thread

**Comment by poolebu** [comment]
Based on comments on the #5637 closed ticket,  I did `yarn upgrade` and when the error appeared, I committed the `yarn.lock` and `package.json`, then ran `jhipster upgrade` again based on @jmlw solution. After that it worked perfectly, thank you! and thank you for this amazing framework

**Comment by jdubois** [comment]
Thanks a lot for your feedback !

**Comment by jdubois** [comment]
Closed by #5883
